app/serializers/detailed_tag_serializer.rb

```diff
@@ -10,11 +10,11 @@ def synonyms
   end
 
   def categories
-    Category.secured(scope).where(id: category_ids)
+    object.all_categories(scope)
   end
 
   def category_restricted
-    !category_ids.empty?
+    !object.all_category_ids.empty?
```

Comment:
How about "reversing" this condition?

```suggestion
    object.all_category_ids.present?
```